### PR TITLE
MicroBit Capacitive Touch

### PR DESCRIPTION
**Pull request by epeach**
Implements the Capacitive Touch functionality for the MicroBit. See API spec: https://docs.google.com/document/d/1oPqJpIsaaGIQv7_KvCShmw6i0mT6MP3VqBP41yZ5M5c/edit#heading=h.ak61axyp722j

Includes tests.

Includes droplet blocks in `Maker` toolbox.

Should only exist behind microbit experiment flag and droplet blocks only appear when using MicroBit and not CP.

## Links

- [spec]()
- [jira]()

## Testing story

# Reviewer Checklist:

- [ ] Tests provide adequate coverage
- [ ] Code is well-commented
- [ ] New features are translatable or updates will not break translations
- [ ] Relevant documentation has been added or updated
- [ ] User impact is well-understood and desirable
- [ ] Pull Request is labeled appropriately
- [ ] Follow-up work items (including potential tech debt) are tracked and linked

### CODE REVIEW

File: apps/src/lib/kits/maker/boards/microBit/CapacitiveTouchSensor.js [hunk 1]
```diff
@@ -0,0 +1,46 @@
+import {EventEmitter} from 'events';
+
+export default class CapacitiveTouchSensor extends EventEmitter {
+  constructor(board) {
+    super();
+    this.board = board;
+
+    // Flag to only trigger event on first of type
```
Comment (ajpal) [inline apps/src/lib/kits/maker/boards/microBit/CapacitiveTouchSensor.js]: Can you clarify this comment?

File: apps/src/lib/kits/maker/boards/microBit/CapacitiveTouchSensor.js [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+    this.connect = false;
```
Comment (ajpal) [inline apps/src/lib/kits/maker/boards/microBit/CapacitiveTouchSensor.js]: I read `connect` as a verb. Would `isConnected` or `hasConnected` make sense here?

File: apps/src/lib/kits/maker/boards/microBit/CapacitiveTouchSensor.js [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+      isPressed: {
+        get: function() {
+          return this.connect;
+        }
+      }
+    });
+  }
+
+  start() {
+    this.board.mb.streamAnalogChannel(this.board.pin); // enable temp sensor
```
Comment (ajpal) [inline apps/src/lib/kits/maker/boards/microBit/CapacitiveTouchSensor.js]: temp? Should be captouch? (and below)

File: apps/src/lib/kits/maker/boards/microBit/CapacitiveTouchSensor.js [hunk 4]
```diff
@@ -0,0 +1,46 @@
...
+    this.readSensorTimer = setInterval(() => {
+      let currentValue = this.board.mb.analogChannel[this.board.pin];
+      if (this.board.mb.analogChannel[this.board.pin] !== 255) {
+        if (currentValue > this.releaseReading + 200 && !this.connect) {
```
Comment (ajpal) [inline apps/src/lib/kits/maker/boards/microBit/CapacitiveTouchSensor.js]: why + 200 here and + 100 below?

File: apps/src/lib/kits/maker/boards/microBit/CapacitiveTouchSensor.js (hunk 3 above)
Comment (epeach) [reply apps/src/lib/kits/maker/boards/microBit/CapacitiveTouchSensor.js]: Ahh good catch - thanks!

File: apps/src/lib/kits/maker/boards/microBit/CapacitiveTouchSensor.js [hunk 5]
```diff
@@ -0,0 +1,62 @@
...
+  }
+
+  start() {
+    this.board.mb.streamAnalogChannel(this.board.pin); // enable cap touch sensor
+    this.readSensorTimer = setInterval(() => {
+      let currentValue = this.board.mb.analogChannel[this.board.pin];
+      if (this.board.mb.analogChannel[this.board.pin] !== 255) {
+        if (
+          currentValue > this.releaseReading + this.connectedDelta &&
+          !this.connect
```
Comment (ajpal) [inline apps/src/lib/kits/maker/boards/microBit/CapacitiveTouchSensor.js]: should now be `connected`?

**Review by ajpal — APPROVED** [review]
one comment, but other than that looks great!

**Pull request merged**